sdk/test/logs/batch_log_processor_test.cc

```diff
@@ -0,0 +1,286 @@
+      new std::vector<std::unique_ptr<LogRecord>>);
+
+  auto batch_processor = GetMockProcessor(logs_received, is_shutdown);
+  const int num_logs   = 2048;
+
+  for (int i = 0; i < num_logs; ++i)
+  {
+    auto log = batch_processor->MakeRecordable();
+    log->SetName("Log" + std::to_string(i));
+    batch_processor->OnReceive(std::move(log));
+  }
+
+  // Give some time to export
+  std::this_thread::sleep_for(std::chrono::milliseconds(500));
+
+  batch_processor->ForceFlush();
+
+  EXPECT_EQ(num_logs, logs_received->size());
+  for (int i = 0; i < num_logs; ++i)
+  {
+    EXPECT_EQ("Log" + std::to_string(i), logs_received->at(i)->GetName());
+  }
+
+  // Create some more logs to make sure that the processor still works
+  for (int i = 0; i < num_logs; ++i)
+  {
+    auto log = batch_processor->MakeRecordable();
+    log->SetName("Log" + std::to_string(i));
+    batch_processor->OnReceive(std::move(log));
+  }
+
+  // Give some time to export the logs
+  std::this_thread::sleep_for(std::chrono::milliseconds(50));
+
+  batch_processor->ForceFlush();
+
+  EXPECT_EQ(num_logs * 2, logs_received->size());
+  for (int i = 0; i < num_logs * 2; ++i)
+  {
+    EXPECT_EQ("Log" + std::to_string(i % num_logs), logs_received->at(i)->GetName());
+  }
+}
+
+TEST_F(BatchLogProcessorTest, TestManyLogsLoss)
+{
+  /* Test that when exporting more than max_queue_size logs, some are most likely lost*/
+
+  std::shared_ptr<std::atomic<bool>> is_shutdown(new std::atomic<bool>(false));
+  std::shared_ptr<std::vector<std::unique_ptr<LogRecord>>> logs_received(
+      new std::vector<std::unique_ptr<LogRecord>>);
+
+  const int max_queue_size = 4096;
+
+  auto batch_processor = GetMockProcessor(logs_received, is_shutdown);
+
+  // Create max_queue_size log records
+  for (int i = 0; i < max_queue_size; ++i)
+  {
+    auto log = batch_processor->MakeRecordable();
+    log->SetName("Log" + std::to_string(i));
+    batch_processor->OnReceive(std::move(log));
+  }
+
+  // Give some time to export the logs
+  std::this_thread::sleep_for(std::chrono::milliseconds(700));
+
+  batch_processor->ForceFlush();
+
+  // Log should be exported by now
+  EXPECT_GE(max_queue_size, logs_received->size());
+}
+
+TEST_F(BatchLogProcessorTest, TestManyLogsLossLess)
+{
+  /* Test that no logs are lost when sending max_queue_size logs */
+
+  std::shared_ptr<std::atomic<bool>> is_shutdown(new std::atomic<bool>(false));
+  std::shared_ptr<std::vector<std::unique_ptr<LogRecord>>> logs_received(
+      new std::vector<std::unique_ptr<LogRecord>>);
+  auto batch_processor = GetMockProcessor(logs_received, is_shutdown);
+
+  const int num_logs = 2048;
+
+  for (int i = 0; i < num_logs; ++i)
+  {
+    auto log = batch_processor->MakeRecordable();
+    log->SetName("Log" + std::to_string(i));
+    batch_processor->OnReceive(std::move(log));
+  }
+
+  // Give some time to export the logs
+  std::this_thread::sleep_for(std::chrono::milliseconds(50));
```

Comment:
IIUC:  You should NOT need any sleep if you `ForceFlush` and your multithreaded code is correct.    I'd remove all the "sleep" code in your unit tests.  If they aren't passing, it's likely a synchronizations/concurrency issue that needs to be fixed.

Reply:
Right, I based it off the way te batch span processor tests were written, but I just removed them and looks like nothing is breaking. I'll definitely look more into optimizing this processor in a separate PR and after testing it more

Reply:
Yup, you're right, the `ForceFlush()` calls should not require this extra sleep time, since the `ForceFlush()` method needs to wait for the worker thread to signal back that its flushing has completed before returning `true`. Removed these from the tests. 

In the last test,`TestScheduleDelayMillis` - I added a comment why it needed that small delay is necessary, since there is no _mandatory_  requirement for  the worker thread (which is performing asynchronously) to complete the current export, before proceeding to the next line of the tests